src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -848,15 +853,23 @@ export default class AircraftModel {
      * @return {boolean}
      */
     isBeyondTopOfDescent() {
-        const waypointModel = this.fms.nextHardAltitudeRestrictedWaypoint;
-
-        if (_isNil(waypointModel)) {
+        if (_isEmpty(this.fms.waypoints)) {
             return;
         }
 
-        const waypointAltitude = waypointModel.altitudeMaximum;
-        const waypointDistance = this.positionModel.distanceToPosition(waypointModel.positionModel);
-        const altitudeChange = waypointAltitude - this.altitude;
+        const currentAltitude = this.altitude;
+        const waypoints = this.fms.getAltitudeRestrictedWaypoints();
+        const waypointsWithRelevantCeiling = _filter(waypoints, (waypoint) => waypoint.hasMaximumAltitudeBelow(this.altitude));
```

Comment:
Wow, you're right, we don't have a `LegCollection`.  That surprised me very much.

Reply:
My opinion is the really proper way of doing it is to encapsulate effectively all the logic within the Fms because minimum altitudes are also a factor.  I'm unclear if you would like a change within this pull request or if it's more for the future refactor.

Reply:
I agree it should be moved out of the aircraft model. But no, I only meant these as thoughts for the future. This is good for now.